src/patch-place-break/patch-place-break-core/src/test/java/fr/djaytan/minecraft/jobsreborn/patchplacebreak/core/PatchPlaceBreakImplTest.java

```diff
@@ -0,0 +1,326 @@
+
+      // When
+      patchPlaceBreakImpl.moveTags(blocks, direction);
+
+      // Then
+      verify(tagRepository, timeout(1000)).updateLocations(locationPairCaptor.capture());
+      OldNewBlockLocationPairSet actualValue = locationPairCaptor.getValue();
+
+      OldNewBlockLocationPairSet expectedValue =
+          new OldNewBlockLocationPairSet(
+              blocks.stream()
+                  .filter(block -> !restrictedBlocksProperties.isRestricted(block))
+                  .map(
+                      oldBlockLocation ->
+                          new OldNewBlockLocationPair(
+                              oldBlockLocation.getBlockLocation(),
+                              BlockLocation.from(oldBlockLocation.getBlockLocation(), direction)))
+                  .collect(Collectors.toSet()));
+
+      assertThat(actualValue).isEqualTo(expectedValue);
+    }
+  }
+
+  @Nested
+  @DisplayName("Is place and break exploit")
+  @TestInstance(Lifecycle.PER_CLASS)
+  class IsPlaceAndBreakExploit {
+
+    @ParameterizedTest(name = "{index} - {0}")
+    @MethodSource
+    @DisplayName("With nominal values")
+    void withUnrestrictedBlock(Tag givenTag, boolean expectedValue) {
+      // Given
+      Block restrictedBlock = Block.of(BlockLocation.of("world", 0, 0, 0), "BEACON");
+      given(tagRepository.findByLocation(any())).willReturn(Optional.ofNullable(givenTag));
+
+      // When
+      boolean isExploit =
+          patchPlaceBreakImpl.isPlaceAndBreakExploit(BlockActionType.BREAK, restrictedBlock);
+
+      // Then
+      assertThat(isExploit).isEqualTo(expectedValue);
+    }
+
+    private @NonNull Stream<Arguments> withUnrestrictedBlock() {
+      return Stream.of(
+          Arguments.of(
+              Named.of(
+                  "With nominal values",
+                  Tag.of(BlockLocation.of("world", 0, 0, 0), false, LocalDateTime.now())),
+              true),
+          Arguments.of(Named.of("With no tag", null), false),
+          Arguments.of(
+              Named.of(
+                  "With active ephemeral tag",
+                  Tag.of(
+                      BlockLocation.of("world", 0, 0, 0),
+                      true,
+                      LocalDateTime.now().minusSeconds(1))),
+              true),
+          Arguments.of(
+              Named.of(
+                  "With inactive ephemeral tag",
+                  Tag.of(
+                      BlockLocation.of("world", 0, 0, 0),
+                      true,
+                      LocalDateTime.now().minusSeconds(10))),
+              false));
+    }
+
+    @Test
+    @DisplayName("With restricted block")
+    void withRestrictedBlock() {
+      // Given
+      Block restrictedBlock = Block.of(BlockLocation.of("world", 0, 0, 0), "STONE");
+
+      // When
+      boolean isExploit =
+          patchPlaceBreakImpl.isPlaceAndBreakExploit(BlockActionType.BREAK, restrictedBlock);
+
+      // Then
+      assertThat(isExploit).isFalse();
```

Comment:
Important! Check interactions with the tagrepository (i.e. verifyNoInteractions with Mockito)